new contribs from aw and database migration

cblgh opened the pull request:
o/ here's the mergin pr, basing off of yours @alexwennerberg. 

there are changes from #53:
* leaving out the "upgrade golang" commit because it forces a higher version of golang than strictly necessary. the `toolchain` keyword was introduced in `1.21` (i think?), and i'm sure it's great, but it was preventing me from building locally so i left it out. honestly not a fan of blind "bump deps" prs either because, semver be damned, people introduce breaking changes anyway in what is described as a patch-bump and that shit sucks to debug. warranted bumps (see next) are 🆗 
* consequently i redid the sql version bump that was prohibiting you from building with musl tooling (redid it bc just using that commit broke, i think cause it was basing off of the changes in `upgrade golang`? not sure)
* i left out the "[Add replies to posts](https://github.com/cblgh/cerca/pull/53/commits/8f5010879a14c170ccf8887ffa399087d5755f9b)" commit for now because i think i'd like to take a stab at as well to search for a version of it that pairs nicely with cerca-as-deployed for the merv forum. the implementation in the commit adds a button for every reply, which aren't hidden when not logged in, creates quote pyramids, embeds the chosen language of the quoter
* the database migration is now written to get the database into a new state following the underlying password lib switch (commits 68a689612547ff83225f9a2727cf0c14dfbf7ceb 27c6d5684b6b464b900889c4b8a4dbae232d6b68); see `MIGRATIONS.md` (pls take a db backup before migrating, just to be safe, but i've tested it in dev and works for me)

hope it's fine that i went in and did a little snip-snip to yr pr! if you're fine with these changes i'll merge them into main and we can continue the journey from there instead of this increasingly long branch :~

alexwennerberg commented:
works for me! the go upgrade is I think just a toolchain issue on my end. go keeps yelling at me to upgrade `go.mod`, I'll figure how how to get it to accept an older pinned `go version` :)

I'll await your `reply` PR. happy to collab on any of those changes as well

cblgh commented:
@alexwennerberg i've rebased my own changes now. while looking things over i noticed the edit timestamp was using publish instead in the template so i fixed that! i also tweaked the styling of edit timestamp and the edit view (just fyi in case it negatively affects :fish:)
![10763gHr](https://github.com/cblgh/cerca/assets/3862362/374056d6-6165-40de-9cba-8b18536af888)
![10764qaN](https://github.com/cblgh/cerca/assets/3862362/cf02e7bf-a492-49cc-b7c2-2efb01200d4e)

LETS GOOO
